src/rust/engine/process_execution/src/docker.rs

```diff
@@ -435,24 +444,18 @@ impl<'a> super::CommandRunner for CommandRunner<'a> {
 
 #[async_trait]
 impl<'a> CapturedWorkdir for CommandRunner<'a> {
-  type WorkdirToken = String;
+  type WorkdirToken = (String, String);
 
   async fn run_in_workdir<'s, 'c, 'w, 'r>(
     &'s self,
-    context: &'c Context,
+    _context: &'c Context,
     _workdir_path: &'w Path,
-    sandbox_path_in_container: Self::WorkdirToken,
+    (container_id, sandbox_path_in_container): Self::WorkdirToken,
```

Comment:
Out of curiosity, why is this a pair and not two arguments? Seems fine, but not obvious.

Reply:
Because it's a generic argument: `Self::WorkdirToken` is defined by the `trait` (think `interface`) that we're implementing.